frb_example/benches/dart/lib/async_benchmark.dart

```diff
@@ -0,0 +1,158 @@
+export 'async_benchmark.io.dart' if (dart.library.html) 'async_benchmark.web.dart' show AsyncBenchmark;
+
+import 'package:flutter_rust_bridge_benchmark/bridge_definitions.dart';
+import 'package:flutter_rust_bridge_benchmark/constants.dart';
+import 'package:flutter_rust_bridge_benchmark/env/stub.dart' show itemsCount, outputFilename;
+
+import 'walltime.dart';
+
+extension UnitSymbol on Unit {
+  String symbol() {
+    switch (this) {
+      case Unit.Milliseconds:
+        return 'ms';
+      case Unit.Microseconds:
+        return 'μs';
+      case Unit.Nanoseconds:
+        return 'ns';
+    }
+  }
+}
+
+class Routine {
+  final int iterations;
+  final double elapsed;
+  const Routine(this.iterations, this.elapsed);
+}
+
+class Sample {
+  final List<Routine> routines;
+  final Unit unit;
+  final String name;
+  const Sample(this.routines, this.name, this.unit);
+  int get iterations => routines.fold(0, (previousValue, element) => previousValue + element.iterations);
+  double get elapsed => routines.fold(0, (previousValue, element) => previousValue + element.elapsed);
+  Map<String, dynamic> toJson() {
+    return {
+      'value': elapsed / iterations,
+      'unit': unit.symbol(),
+      'name': name,
+      'extra': itemsCount.toString(),
+    };
+  }
+}
+
+abstract class Bencher {
+  final String name;
+  final String dylibPath;
+  final Duration warmUpTime;
+  final Duration measurementTime;
+  final int sampleSize;
+
+  WallTime start();
+
+  Bencher(
+      {required this.name,
+      required this.dylibPath,
+      required this.warmUpTime,
+      required this.measurementTime,
+      required this.sampleSize});
+
+  Future<void> setup() {
+    return Future.sync(() {
+      print(
+          'setup (sample count: $sampleSize, items count: $itemsCount, warm up time: ${warmUpTime.inMilliseconds} ${Unit.Milliseconds.symbol()}, measurement time: ${measurementTime.inMilliseconds} ${Unit.Milliseconds.symbol()})');
+    });
+  }
+
+  Future<void> teardown();
+
+  /// see [criterion warmup](https://bheisler.github.io/criterion.rs/book/analysis.html#warmup)
+  Future<Routine> warmup(Future<void> Function() f);
+
+  /// see [criterion warmup](https://bheisler.github.io/criterion.rs/book/analysis.html#measurement)
+  Future<Sample> measure(Future<void> Function() f, double warmUpElapsed, int warmUpIter, int sampleSize);
+
+  Future<void> run();
+
+  Future<Sample> bench() async {
+    await setup();
+    try {
+      final routine = await warmup(run);
+      return await measure(run, routine.elapsed, routine.iterations, sampleSize);
```

Comment:
uber-nit: Maybe we should not put `run` as an argument, but should directly call it (`this.run`)? Because I am afraid dart compiler is not that smart to inline the `run` closure